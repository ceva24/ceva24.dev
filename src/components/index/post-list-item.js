import React from "react";
import { css } from "@emotion/react";
import { rhythm } from "src/utils/typography";
import { Link } from "gatsby";
import PostDate from "src/components/post-date";

const PostListItem = ({ node }) => (
    <li
        key={node.id}
        css={css`
            margin-bottom: ${rhythm(1)};
        `}
    >
        <div
            css={css`
                display: flex;
            `}
        >
            <h3
                css={css`
                    margin-bottom: ${rhythm(1 / 3)};
                `}
            >
                <Link
                    to={`/${node.fields.path}`}
                    css={css`
                        text-decoration: none;
                    `}
                >
                    {node.frontmatter.title}
                </Link>
            </h3>
            <div
                css={css`
                    flex-grow: 2;
                    text-align: right;
                    padding-left: ${rhythm(3 / 4)};
                `}
            >
                <PostDate>{node.frontmatter.date}</PostDate>
            </div>
        </div>
        <p>{node.excerpt}</p>
    </li>
);

export default PostListItem;
