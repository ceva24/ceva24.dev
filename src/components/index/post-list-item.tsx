import { css } from "@emotion/react";
import { Link } from "gatsby";
import { rhythm } from "../../utils/typography";
import { PostDate } from "../post-date";

interface Node { 
    node: {
        id: string,
        fields: {
            path: string
        },
        frontmatter: {
            title: string,
            date: string
        },
        excerpt: string
    }
}

const PostListItem: React.FC<Node> = ({ node }: Node) => (
    <li
        key={node.id}
        css={css`
            margin-bottom: ${rhythm(1)};
        `}
    >
        <div
            css={css`
                display: flex;
            `}
        >
            <h3
                css={css`
                    margin-bottom: ${rhythm(1 / 3)};
                `}
            >
                <Link
                    to={`/${node.fields.path}`}
                    css={css`
                        text-decoration: none;
                    `}
                >
                    {node.frontmatter.title}
                </Link>
            </h3>
            <div
                css={css`
                    flex-grow: 2;
                    text-align: right;
                    padding-left: ${rhythm(3 / 4)};
                `}
            >
                <PostDate>{node.frontmatter.date}</PostDate>
            </div>
        </div>
        <p>{node.excerpt}</p>
    </li>
);

export default PostListItem;
