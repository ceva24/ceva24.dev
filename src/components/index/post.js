import React from "react"
import { css } from "@emotion/core"
import { rhythm } from "src/utils/typography"
import {Link} from "gatsby";

export default ({node}) => (
    <div key={node.id}>
        <h3 css={css`
            margin-bottom: ${rhythm(1 / 4)};
        `}>
            <Link to={node.fields.path} css={css`
                text-decoration: none;
            `}>
                {node.frontmatter.title}{" "}
            </Link>
            <span css={css`
                color: #7f8c8d;
            `}>
                â€” {node.frontmatter.date}
            </span>
        </h3>
        <p>{node.excerpt}</p>
    </div>
)
