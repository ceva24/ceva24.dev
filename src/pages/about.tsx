import { StaticImage } from "gatsby-plugin-image";
import React from "react";
import { Layout } from "../components/layout";

const PureAbout: React.FC = () => (
    <>
        <h2>About me</h2>

        <p>
            Hi! I&apos;m a Development Team Lead from York, UK. I currently work
            at the University of York, leading a team of developers building
            great products and maintaining the systems that support the
            outstanding teaching and learning delivered by the University.
        </p>

        <div className="max-w-lg mx-auto mb-5">
            <figure>
                <StaticImage
                    src="../img/chris-evans.jpg"
                    alt="Chris Evans"
                    quality={90}
                    className="border-4"
                />
                <figcaption className="italic text-gray-500">
                    I should really find a more up-to-date picture
                </figcaption>
            </figure>
        </div>

        <p>
            I started out in Java and then moved into back-end web development
            with Groovy/Grails. A while back I transitioned to
            JavaScript/TypeScript, Node.js and React (mainly Next.js and
            Gatsby). I&apos;ve also built up a chunk of experience in AWS
            Serverless and event-driven architecture.
        </p>

        <p>
            I&apos;m really keen on automation and using it to enhance
            teams&apos; capabilities - I take full advantage of GitHub workflows
            both at work and in personal projects!
        </p>

        <p>
            I&apos;m also interested in agile team leadership - driving change
            by focusing on user benefit, breaking things down into deliverable
            increments and using metrics to inform success. I value an open and
            collaborative approach to software development, always asking the
            question &quot;what value are we delivering?&quot;
        </p>

        <p>
            I&apos;m passionate about people - attracting and recruiting the
            best talent, doing onboarding right, and supporting developers&apos;
            careers to drive them towards success.
        </p>

        <p>
            Please don&apos;t hesitate to{" "}
            <a href="mailto:chris@ceva24.dev">reach out to me</a> for a chat, to
            ask a question, or anything else really!
        </p>

        <h2>About this blog</h2>

        <p>
            I started this blog back in 2012 to have a bit more of a web
            presence, and to use it as a place to drop thoughts, code snippets,
            and anything else I wanted to talk about (hence the tagline
            &quot;Thoughts, code and everything in-between&quot;). It&apos;s
            fair to say I haven&apos;t made the most out of it, but I still like
            to update it every so often.
        </p>

        <p>Thanks for taking the time to check it out!</p>
    </>
);

const About: React.FC = () => {
    return (
        <Layout showFooter>
            <PureAbout />
        </Layout>
    );
};

export default About;
export { PureAbout };
