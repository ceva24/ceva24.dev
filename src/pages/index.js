import React from "react"
import { graphql } from "gatsby"
import { css } from "@emotion/core"
import Layout from "src/components/layout"
import PostListItem from "src/components/index/post-list-item"

export default ({ data }) => (
    <Layout>
        <h2>Posts</h2>
        <ul css={css`
            list-style: none;
            margin-left: 0;
        `}>
            {data.allMarkdownRemark.edges.map(({node}) => (
                <PostListItem node={node} />
            ))}
        </ul>
    </Layout>
)

export const query = graphql`
  query {
    allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
      totalCount
      edges {
        node {
          id
          frontmatter {
            title
            date(formatString: "YYYY-MM-DD")
          }
          fields {
            slug
            path
          }
          excerpt
        }
      }
    }
  }
`
