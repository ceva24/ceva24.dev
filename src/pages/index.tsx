import React from "react";
import { graphql } from "gatsby";
import { Layout } from "../components/layout";
import { PostListItem } from "../components/post-list-item";

const Index: React.FC<IndexPageData> = ({ data }: IndexPageData) => {
    return (
        <Layout
            title={`${data.site.siteMetadata.name} | ${data.site.siteMetadata.subtitle}`}
        >
            <h2>
                Posts <span>({data.allMarkdownRemark.totalCount})</span>
            </h2>
            <ul>
                {data.allMarkdownRemark.edges.map((edge) => (
                    <PostListItem key={edge.node.id} node={edge.node} />
                ))}
            </ul>
        </Layout>
    );
};

// eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
const query = graphql`
    query {
        site {
            siteMetadata {
                name
                subtitle
            }
        }
        allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
            totalCount
            edges {
                node {
                    id
                    frontmatter {
                        title
                        date(formatString: "YYYY-MM-DD")
                    }
                    fields {
                        slug
                        path
                    }
                    excerpt
                }
            }
        }
    }
`;

export default Index;
export { query };
