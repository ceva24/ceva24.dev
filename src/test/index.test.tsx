import React from "react";
import { render } from "@testing-library/react";
import { PureIndex } from "../pages/index";

jest.mock("gatsby", (): unknown => {
    return {
        ...jest.requireActual("gatsby"),
        graphql: jest.fn(),
        StaticQuery: jest.fn(),
        useStaticQuery: jest.fn(),
    };
});

jest.mock("@gatsbyjs/reach-router", (): unknown => ({
    ...jest.requireActual("@gatsbyjs/reach-router"),
    useLocation: jest.fn().mockImplementation(() => {
        return { pathname: "/" };
    }),
}));

describe("index", () => {
    it("renders", () => {
        const data: IndexPageData = {
            data: {
                allMarkdownRemark: {
                    totalCount: 1,
                    edges: [
                        {
                            node: {
                                id: "abc",
                                frontmatter: {
                                    title: "Post title",
                                    date: "2000-01-01",
                                },
                                fields: {
                                    slug: "post-title",
                                    path: "posts/post-title",
                                },
                                excerpt: "This is a post...",
                            },
                        },
                    ],
                },
            },
        };

        const { asFragment } = render(<PureIndex {...data} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
