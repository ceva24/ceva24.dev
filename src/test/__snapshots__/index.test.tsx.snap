// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post renders 1`] = `
<DocumentFragment>
  <ul
    class="list-none"
  >
    <li
      aria-label="Post title"
      class="mb-0"
    >
      <a
        aria-label="Post title"
        class="text-black"
        href="/posts/post-title"
      >
        <div
          class="p-6 pb-3 hover:bg-gray-50 hover:shadow-lg"
        >
          <div
            class="flex gap-4"
          >
            <h2
              class="flex-auto text-xl uppercase font-normal"
            >
              Post title
            </h2>
            <div
              class="flex-auto text-right"
            >
              2000-01-01
            </div>
          </div>
          <p>
            This is a post...
          </p>
        </div>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
